contracts/router/src/contract/execute.rs

```diff
@@ -208,19 +208,20 @@ pub fn route_messages(
         .group_by(|msg| msg.destination_chain.to_owned())
         .into_iter()
         .map(|(destination_chain, msgs)| {
-            let gateway = match state::load_chain_by_chain_name(storage, &destination_chain)? {
-                Some(destination_chain) if destination_chain.outgoing_frozen() => {
-                    return Err(report!(Error::ChainFrozen {
-                        chain: destination_chain.name,
-                    }));
-                }
-                Some(destination_chain) => destination_chain.gateway.address,
-                // messages with unknown destination chains are routed to
-                // the nexus gateway if the sender is not the nexus gateway
-                // itself
-                None if sender != config.nexus_gateway => config.nexus_gateway.clone(),
-                _ => return Err(report!(Error::ChainNotFound)),
-            };
+            let gateway =
+                match state::load_chain_by_chain_name(storage, &destination_chain.amplifier()?)? {
```

Comment:
Is this line right? Isn't the `?` going to cause this whole closure to return an error if the chain is a legacy chain, instead of routing to nexus?